burr/core/application.py

```diff
@@ -1041,6 +1078,23 @@ def __init__(self):
         self.actions: Optional[List[Action]] = None
         self.start: Optional[str] = None
         self.lifecycle_adapters: List[LifecycleAdapter] = list()
+        self.app_id: str = str(uuid.uuid4())
```

Comment:
Put this in `build`, not `__init__` -- this restricts a single builder to a single `app`, which is not required. Also its weird to set it to a value that has meaning then overwrite.